### PR TITLE
Make Captions Setting Modal a public component

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
Make Captions Setting Modal a public component

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by carocao-msft — COMMENTED** [review]
Potential breaking change for calling component binding, need to discuss in stand up

**Review by edwardlee-msft — APPROVED** [review]

**Review by mgamis-msft — APPROVED** [review]

File: packages/storybook8/stories/Components/CaptionsSettingsModal/CaptionsSettingsModal.story.tsx [hunk 1]
```diff
@@ -53,31 +53,40 @@ const CaptionsSettingsModalStory = (args: any): JSX.Element => {
...
+  ];
   const currentSpokenLanguage = 'en-us';
+  const currentCaptionLanguage = 'en';
   const [showModal, setShowModal] = useState<boolean>(false);
   const onDismissCaptionsSettings = (): void => {
     setShowModal(false);
   };
   const onSetSpokenLanguage = (language: any): Promise<void> => {
-    alert(`Selected ${language}`);
+    console.log(`Spoken language set to ${language}`);
```
Comment (emlynmac) [inline packages/storybook8/stories/Components/CaptionsSettingsModal/CaptionsSettingsModal.story.tsx:79]: Let's remove console.logs

Comment (carocao-msft) [reply packages/storybook8/stories/Components/CaptionsSettingsModal/CaptionsSettingsModal.story.tsx:79]: This is to make the storybook page a better experience. Alert can be jarring while console.log is quieter.

**Review by alkwa-msft — APPROVED** [review]

File: packages/storybook8/stories/Components/CaptionsSettingsModal/CaptionsSettingsModal.story.tsx (hunk 1 above)
Comment (carocao-msft) [reply packages/storybook8/stories/Components/CaptionsSettingsModal/CaptionsSettingsModal.story.tsx]: Updated to use alert instead

**Review by carocao-msft — COMMENTED** [review]
Bundle size change caused by additional locale strings

**Pull request merged**